### PR TITLE
Zig 0.11

### PR DESCRIPTION
**Pull request by YellowApple**
## Description

Fixes some compilation errors I encountered when building under Zig 0.11.0-dev.2287+1de64dba2.  ~~This PR probably ain't ready to merge yet, since there are also some errors in `zig-clap` that require a separate PR there (and some git submodule shenanigans, probably).~~ `zig-clap` is already Zig-0.11-ready on `master`, so that's good to go.

## Motivation and Context

This is required to build `linuxwave` with Zig versions 0.11 and greater.

## How Has This Been Tested?

```shell
zig version  # outputs 0.11.0-dev.2287+1de64dba2
zig build -Doptimize=ReleaseSafe  # to make sure the build process still works in isolation
zig build test
zig build run -- -s 0,7,1 -n 60 -v 65 -o boss.wav  # and other "presets" from the README
```

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other 

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by orhun** [comment]
Hello! 🐻

Thanks for your contribution. However, I'm not sure if switching to Zig master is something that will benefit the project right now. Most of the prominent Linux distros have the latest _stable_ version of Zig (as of `0.10.1`) packaged in their repositores and switching to `0.11.0-dev.2287+1de64dba2` will mean that they won't be able to distribute `linuxwave`. Also, it will make the manual compilation harder for users since they will need the development version of Zig and it might have some quirks in the future.

I will keep this PR open until there is a stable release for 0.11 and we can merge it when the adoption is becoming higher.

File: build.zig [hunk 1]
```diff
@@ -9,16 +9,26 @@ const version = "0.1.2"; // managed by release.sh
...
 ///
 /// This is used for providing the dependencies for main executable as well as the tests.
-fn addPackages(allocator: std.mem.Allocator, exe: *std.build.LibExeObjStep) !void {
-    exe.addPackagePath("clap", "libs/zig-clap/clap.zig");
-    for ([_][]const u8{ "file", "gen", "wav" }) |package| {
-        const path = try std.fmt.allocPrint(allocator, "src/{s}.zig", .{package});
-        defer allocator.free(path);
-        exe.addPackagePath(package, path);
-    }
+fn addPackages(b: *std.Build, exe: *std.build.LibExeObjStep) !void {
```
Comment (kassane) [inline build.zig:12]: why not `*std.Build.CompileStep`?

File: build.zig [hunk 2]
```diff
@@ -9,16 +9,26 @@ const version = "0.1.2"; // managed by release.sh
...
+        .source_file = .{ .path = "src/wav.zig" },
+        .dependencies = &.{},
+    }));
 }
 
-pub fn build(b: *std.build.Builder) !void {
+pub fn build(b: *std.Build) !void {
     // Create an allocator.
     var gpa = std.heap.GeneralPurposeAllocator(.{}){};
     const allocator = gpa.allocator();
```
Comment (kassane) [inline build.zig:34]: why not using `b.allocator`?

**Review by kassane — CHANGES_REQUESTED** [review]

File: build.zig (hunk 2 above)
Comment (orhun) [reply build.zig:34]: applied to main in 1b84f60e203600078d9a778979f540d5ade57019

File: build.zig [hunk 3]
```diff
@@ -73,19 +86,24 @@ pub fn build(b: *std.build.Builder) !void {
     // Add tests.
     const test_step = b.step("test", "Run tests");
     for ([_][]const u8{ "main", "wav", "file", "gen" }) |module| {
+        const test_name = try std.fmt.allocPrint(allocator, "{s}-tests", .{module});
+        defer allocator.free(test_name);
         const test_module = try std.fmt.allocPrint(allocator, "src/{s}.zig", .{module});
```
Comment (kassane) [inline build.zig:91]: why not `b.fmt(...)`? No need allocator!

File: build.zig [hunk 4]
```diff
@@ -51,7 +64,7 @@ pub fn build(b: *std.build.Builder) !void {
     exe.install();
```
Comment (kassane) [inline build.zig:64]: Lastest zig (0.11.0-dev) update, replacing:
* `exe.install()` to `b.installArtifact(exe)` to build
* `exe.run()` to `b.addRunArtifact(exe)` to run

**Review by kassane — CHANGES_REQUESTED** [review]
more suggestions

File: build.zig (hunk 3 above)
Comment (orhun) [reply build.zig:91]: applied to main in ef79489af42737b6d3e8d6d72b5de514da9f38c9

**Comment by YellowApple** [comment]
Yeah, I definitely don't expect this to be merged before 0.11's actually released; more just documenting the steps I needed to take to get it to build.

**Comment by orhun** [comment]
I see, thank you for making that clear and submitting this PR!

**Comment by booniepepper** [comment]
Zig 0.11.0 is now released! 🎉

**Comment by orhun** [comment]
Ping @YellowApple, can you look into this once again?

**Comment by orhun** [comment]
done in #28 

thanks for your efforts everyone!

**Pull request closed without merging**